modules/appeals_api/spec/workers/appeal_received_job_spec.rb

```diff
@@ -0,0 +1,219 @@
+              personalisation: {
+                'first_name' => 'first name',
+                'date_submitted' => 'January 02, 2021'
+              }
+            }
+          )
+        end
+      end
+
+      it 'does not care about the order of email identifier hash' do
+        with_settings(Settings.vanotify.services.lighthouse.template_id,
+                      higher_level_review_received: 'veteran_template',
+                      higher_level_review_received_claimant: 'claimant_template') do
+          client = instance_double(VaNotify::Service)
+          allow(VaNotify::Service).to receive(:new).and_return(client)
+          allow(client).to receive(:send_email)
+
+          opts = {
+            'receipt_event' => 'hlr_received',
+            'email_identifier' => { 'id_type' => 'email', 'id_value' => 'fake_email@email.com' }, # key order changed
+            'first_name' => 'first name',
+            'date_submitted' => DateTime.new(2021, 1, 2, 3, 4, 5).iso8601,
+            'guid' => '1234556'
+          }
+
+          described_class.new.perform(opts)
+
+          expect(client).to have_received(:send_email).with(
+            {
+              email_address: 'fake_email@email.com',
+              template_id: 'veteran_template',
+              personalisation: {
+                'first_name' => 'first name',
+                'date_submitted' => 'January 02, 2021'
+              }
+            }
+          )
+        end
+      end
+
+      it 'sends email to claimant using the claimant template' do
+        with_settings(
+          Settings.vanotify.services.lighthouse.template_id,
+          higher_level_review_received: 'veteran_template',
+          higher_level_review_received_claimant: 'claimant_template'
+        ) do
+          client = instance_double(VaNotify::Service)
+          allow(VaNotify::Service).to receive(:new).and_return(client)
+          allow(client).to receive(:send_email)
+
+          opts = {
+            'receipt_event' => 'hlr_received',
+            'email_identifier' => { 'id_type' => 'email', 'id_value' => 'fake_email@email.com' }, # key order changed
+            'first_name' => 'veteran first name',
+            'date_submitted' => DateTime.new(2021, 1, 2, 3, 4, 5).iso8601,
+            'guid' => '1234556',
+            'claimant_email' => 'fake_claimant_email@email.com',
+            'claimant_first_name' => 'Betty'
+          }
+
+          described_class.new.perform(opts)
+
+          expect(client).to have_received(:send_email).with(
+            {
+              email_address: 'fake_claimant_email@email.com',
+              template_id: 'claimant_template',
+              personalisation: {
+                'first_name' => 'Betty',
+                'date_submitted' => 'January 02, 2021',
+                'veterans_name' => 'veteran first name'
+              }
+            }
+          )
+        end
+      end
+
+      it 'raises if claimant info is missing email' do
```

Comment:
Let's move this up to be grouped with the other error checking specs. Also, it's not raising anymore in this circumstance, so the test description should be updated :)